finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -75,7 +75,22 @@ object CanBeParameter {
     new CanBeParameter[BigInt] {
       def sizeOf(param: BigInt) = 8
       def typeCode(param: BigInt) = Type.LongLong
-      def write(writer: MysqlBufWriter, param: BigInt) = writer.writeBytes(param.toByteArray)
+      def write(writer: MysqlBufWriter, param: BigInt) = {
+        val byteArray: Array[Byte] = param.toByteArray
+        val lengthOfByteArray: Int = byteArray.length
+
+        if (lengthOfByteArray > 8) {
+          throw new Exception(s"The length of BigInt is larger than 8 bytes: $lengthOfByteArray > 8")
```

Comment:
Can you create a new named exception for this (can define it somehwere in the file here). I suggest something like:
`
class BigIntTooLongException(size: Int) extends Exception(s"BigInt is stored as an unsigned long and thus cannot be grater than 8 bytes. Size: $size"
`